### PR TITLE
New replay test for buildomat

### PR DESCRIPTION
**Pull request by leftwo**
New replay test for buildomat. This uses crutest and dsc together
to create a better reconnect test than the previous bash crimes that
were how this was tested before.

Remove and delete test_reconnect.sh from usage, it was broken anyway.
Updated the live repair test, less bash more rust.

Fixed a few DTrace scripts that were broken.

Updated the nightly test on to use the new replay test.
Updated several tests to take a "number of loops" options so regular
callers can have the default (shorter) length tests, and the nightly
test can supply a larger number of loops.

### CODE REVIEW

**Comment by leftwo** [comment]
Fixes: https://github.com/oxidecomputer/crucible/issues/822

File: tools/test_live_repair.sh [hunk 1]
```diff
@@ -1,75 +1,71 @@
 #!/bin/bash
 
-# Create and start up the downstairs.
-# In a loop:
-#  Send IO through crutest
-#  Fault a downstairs
-#  Let the upstairs repair.
-
+# Create regions and start up four downstairs.
+# We create four and use the forth downstairs as the replacement
```
Comment (jmpesp) [inline tools/test_live_repair.sh:4]: typo: fourth

File: tools/test_live_repair.sh [hunk 2]
```diff
@@ -1,75 +1,71 @@
...
 
 loop_log=/tmp/test_live_repair_summary.log
 test_log=/tmp/test_live_repair.log
-dsc_test_log=/tmp/test_live_repair_dsc.log
-verify_file=/tmp/test_live_verify
 
 ROOT=$(cd "$(dirname "$0")/.." && pwd)
-export BINDIR=${BINDIR:-$ROOT/target/release}
-cds="$BINDIR/crucible-downstairs"
+export BINDIR=${BINDIR:-$ROOT/target/debug}
```
Comment (jmpesp) [inline tools/test_live_repair.sh:27]: changing to debug here?

File: tools/test_replay.sh [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+trap ctrl_c INT
+function ctrl_c() {
+    echo "Stopping at your request"
+    ${dsc} cmd shutdown
+}
+
+test_log=/tmp/test_replay.log
+
+ROOT=$(cd "$(dirname "$0")/.." && pwd)
+export BINDIR=${BINDIR:-$ROOT/target/debug}
```
Comment (jmpesp) [inline tools/test_replay.sh:22]: debug here?

**Review by jmpesp — APPROVED** [review]
LGTM, but: any way to remove the sleeps?

**Pull request merged**